TOOLS/gen-interface-changes.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+# Generate a new interface-changes.rst based on the entries in
+# the interface-changes directory.
+
+#
+# This file is part of mpv.
+#
+# mpv is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# mpv is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import pathlib
+import sys
+
+def add_new_entries(docs_dir, out):
+    changes_dir = pathlib.Path(docs_dir) / "interface-changes"
+    files = [f for f in pathlib.Path(changes_dir).glob("*.txt") if f.is_file()]
+    files.sort()
+    for file in files:
+        if file.name == "example.txt":
+            continue
+        with open(file.resolve(), "r") as f:
+            for line in f:
+               line =  "    - " + line.rstrip()
+               out.write(line + "\n")
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print(f"Usage: {sys.argv[0]} <version>")
+        sys.exit(1)
+
+    # Accept passing only the major version number and the full 0 version.
+    major_version = -1
+    if sys.argv[1].isdigit():
+        major_version = sys.argv[1]
+    else:
+        ver_split = sys.argv[1].split(".")
+        if len(ver_split) == 3 and ver_split[1].isdigit():
+            major_version = ver_split[1]
+
+    if major_version == -1:
+        print(f"Invalid version number: {sys.argv[1]}")
+        sys.exit(1)
+
+    docs_dir = pathlib.Path(sys.argv[0]).resolve().parents[1] / "DOCS"
+    interface_changes = docs_dir / "interface-changes.rst"
+    with open(interface_changes, "r") as f:
+        lines = [line.rstrip() for line in f]
+
+    ver_line = " --- mpv 0." + major_version + ".0 ---"
```

Comment:
who adds this line? the release manager?
the script could as well do it

Reply:
It previously did, but I ended up removing it because that would mean you would have:
```
 --- mpv 0.39.0 ---
 --- mpv 0.38.0 ---
    - actual changes start here
```
In the 0.38.0 release unless you remove the 0.39.0 line manually. I really don't mind either way tbh; whatever you prefer.

Reply:
script should add it IMO

Reply:
Done.